### PR TITLE
ready to code review

### PR DESCRIPTION
**Pull request by xtarzyk**

### CODE REVIEW

File: circles.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const gameContainer = document.createElement('div')
+gameContainer.classList.add('game-container')
+const circle = document.createElement('div')
+circle.classList.add('game-container__circle')
+let circles = Array.from({length: 5})
+circles.fill(circle)
```
Comment (jpudysz) [inline circles.js:6]: Maybe you can put it into separate function like `prepareUI` or `createGameUI`

File: circles.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const gameContainer = document.createElement('div')
+gameContainer.classList.add('game-container')
+const circle = document.createElement('div')
+circle.classList.add('game-container__circle')
+let circles = Array.from({length: 5})
```
Comment (jpudysz) [inline circles.js:5]: spaces between `{}`

File: circles.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+circles.fill(circle)
+
+
+const createCircles = () => {
+    let id = 0
```
Comment (jpudysz) [inline circles.js:10]: before `let const` and `instructions` add enter

File: circles.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    circles.forEach(() => {
+        const newCircle = circle.cloneNode(true)
+        newCircle.setAttribute('id', `${id}`)
+        id++
```
Comment (jpudysz) [inline circles.js:14]: bad mutation, why can't you use `index`?

File: circles.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        if(Number(circle.getAttribute('id')) === currentMark) {
+            circle.classList.remove('game-container__circle')
+            circle.classList.add('red-circle')
+        }
+    }))
+}
+
+const resetColors = () => {
+    const circlesDivs = document.querySelectorAll('.red-circle, .green-circle')
+    circlesDivs.forEach(circle => {
```
Comment (jpudysz) [inline circles.js:33]: enter before `const` and instructions

File: circles.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    circlesDivs.forEach((circle => {
+        if(Number(circle.getAttribute('id')) === currentMark) {
+            circle.classList.remove('game-container__circle')
+            circle.classList.add('red-circle')
+        }
+    }))
+}
+
+const resetColors = () => {
+    const circlesDivs = document.querySelectorAll('.red-circle, .green-circle')
```
Comment (jpudysz) [inline circles.js:32]: you're doing query selector each time you invoke `resetColors`, thats a bad practice

File: circles.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    circles.forEach(() => {
+        const newCircle = circle.cloneNode(true)
+        newCircle.setAttribute('id', `${id}`)
+        id++
+        gameContainer.appendChild(newCircle)
+    })
+}
+
+const paintCircle = () => {
+    const circlesDivs = document.querySelectorAll('.game-container__circle')
```
Comment (jpudysz) [inline circles.js:20]: same here, check comment below

File: circles.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    const circlesNum = circlesDivs.length
+    const currentMark = Math.floor(Math.random() * circlesNum)
+    circlesDivs.forEach((circle => {
```
Comment (jpudysz) [inline circles.js:23]: unnecessary `()`

File: circles.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        if(Number(circle.getAttribute('id')) === currentMark) {
```
Comment (jpudysz) [inline circles.js:24]: I don't get this if

File: circles.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    circlesDivs.forEach(circle => {
+        circle.classList.remove('red-circle')
+        circle.classList.remove('green-circle')
+        circle.classList.add('game-container__circle')
+    })
+}
+
+createCircles()
+
+export {gameContainer, paintCircle, resetColors, createCircles}
```
Comment (jpudysz) [inline circles.js:42]:
```
export {
    gameContainer, 
    paintCircle,
    resetColors,
    createCircles
}
```

File: main.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import {timer, createTimer} from "./timer"
```
Comment (jpudysz) [inline main.js:1]: single quote

File: main.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import {gameContainer} from "./circles"
```
Comment (jpudysz) [inline main.js:2]: spaces between `{}`

File: package.json [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  "version": "0.0.0",
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview"
+  },
+  "devDependencies": {
+    "vite": "^2.9.0"
+  }
+}
```
Comment (jpudysz) [inline package.json:13]: styles.css and styles.css.map to gitignore

File: timer.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
+import {paintCircle, resetColors} from "./circles"
```
Comment (jpudysz) [inline timer.js:1]: singlequote

File: timer.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+        } else {
+            clearInterval(countDown)
+            alert(`Score: ${scores}`)
+        }
+    }, 1000)
+}
+
+
+export {timer, createTimer}
```
Comment (jpudysz) [inline timer.js:34]: spaces between `{}`

File: timer.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import {paintCircle, resetColors} from "./circles"
+import {scores} from "./main"
+
+
+const timer = document.createElement('div')
+timer.classList.add('timer')
+
+const createTimer = () => {
+    const timerDiv = document.querySelector('.timer')
+    let secondsLeft = 59
```
Comment (jpudysz) [inline timer.js:10]: you can use single timer like: `seconds`

File: timer.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    let minutesLeft = 1
+    const countDown = setInterval(() => {
+        if (minutesLeft) {
```
Comment (jpudysz) [inline timer.js:13]:
```
if (seconds / 60 > 0)
```

File: timer.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+            timerDiv.innerHTML = `Timer: 0${minutesLeft}:00`
+            minutesLeft--
+            resetColors()
+            paintCircle()
+            return
+        }
+        if (secondsLeft >= 0) {
```
Comment (jpudysz) [inline timer.js:20]:
```
if (seconds % 60 >= 0)
```

File: timer.js [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+            secondsLeft < 10 ? timerDiv.innerHTML = `Timer: 00:0${secondsLeft}` : timerDiv.innerHTML = `Timer: 00:${secondsLeft}`
+            secondsLeft -= 1
+            resetColors()
+            paintCircle()
+
+        } else {
```
Comment (jpudysz) [inline timer.js:26]: please avoid `else`

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: circles.js (hunk 4 above)
Comment (xtarzyk) [reply circles.js:14]: true

File: circles.js (hunk 6 above)
Comment (xtarzyk) [reply circles.js:32]: every time the resetColors is invoke it needs to get current state of all circles divs, otherwise it's not works

**Review by xtarzyk — COMMENTED** [review]
done

**Comment by jpudysz** [comment]
You can't merge it. It will be merged after approve

**Comment by jpudysz** [comment]
And I can't see any fixes. Please git push them

**Pull request merged**